### ISSUE TITLE
You Are Here (H2 Issue)

### ISSUE THREAD

**Issue by Robineast** [wet-boew/wet-boew#7432]
@wet-boew/access-working-group This message is useless for Screen readers. Should this be  called Bread Crumb as it was in WET 2? As a JAWS user this is rather insulting as it says I am here and I know where I am but in relation to the page on this screen this message serves no useful information what so ever. If it is "Bread Crumb" as a title we would know it would be "here" just before the Bread Crumb links. This would make sense. You Are Here just is not appropriate.

**Comment by pjackson28** [comment]
@Robineast I'll have to go from memory for this response as I haven't had any luck finding where the discussion around this actually took place.

WET originally had "Breadcrumb trail" but there was a long discussion about this quite a while ago (during the WET4 beta) with numerous people from the external web accessibility community. Their recommendation was to move to "You are here" because "Breadcrumb trail" only made sense if you understood the source context (that a breadcrumb trail is used to find your way home as was done in Hansel and Gretel). They indicated that many screen reader users had difficulty with "Breadcrumb trail" because many lacked the necessary context. They indicated that "You are here" worked much better for those screen reader users and that it was the recommended convention going forward. They also recommended including it in a heading to make it easier to navigate to that breadcrumb as the argument was that the user would go to the "You are here" section to find out where "here" was in relation to the site argument. 

I also recall that during that discussion some of their other arguments were that Drupal was baking "You are here" into it's core for accessibility reasons (Drupal had it's own similar breadcrumb discussion previous to WET's), some web accessibility sites such as webaim.org were using "You are here" and even a lot of regular sites out in the wild were using "You are here" as well. It was also provided in the example in the breadcrumb navigation section in the HTML5 specification which also includes "You are here" in the heading (https://www.w3.org/TR/html5/common-idioms.html#rel-up). In addition, based upon an environmental analysis at the time, the "You are here" convention seemed to be far more common than "Breadcrumb trail" which would mean that users were more likely to be familiar with "You are here" than "Breadcrumb trail".

I also recall that there was also usability testing on the overall WET template done with numerous screen reader users (across numerous screen reader brands) and they didn't seem to encounter any issues with the "You are here" wording (did not seem to negatively impact their ability to use the breadcrumb trail).

**Comment by shawnthompson** [comment]
What about the GCWeb theme and removing the title in the breadcrumb list? Reading "You are here" and then Home, then parent of page if there is one doesn't really work anymore.

[Policing, justice and emergencies](https://www.canada.ca/en/services/policing.html)

**Comment by Robineast** [comment]
@pjackson28 
You Are Here has no meaning for a screen reader. It means nothing. There is no relation to the actual location or position on a a screen. The text itself has no meaning to what is trying to portray. It fails the the following two WCAG compliance:

2.4.6 Headings and Labels: Headings and labels describe topic or purpose. (Level AA)
http://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-descriptive.html 

The intent of this Success Criterion is to help users understand what information is contained in Web pages and how that information is organized. When headings are clear and descriptive, users can find the information they seek more easily, and they can understand the relationships between different parts of the content more easily.

1.3.1 Info and Relationships: Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text. (Level A)
http://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html 

There is nothing like "you are here" in Web Aim. The navigation in HTML5  uses the navtags to identify navigation regions for example on Canada.ca they have navigation regions with an H2 Topics which is meaningful. The H2 You Are Here  under all the other departments / agencies has no real meaning to a user. It would make sense to have an H2 Return To: instead of the current H2 You Are Here as the breadcrumb is a trail of hyperlinks that lead consecutively to the current page.
 
A

**Comment by pjackson28** [comment]
@Robineast There are others in the accessibility community that have a different opinion. I personally was in favour of "Breadcrumb trail" but ultimately the project went with the consensus opinion of those in the accessibility community that participated in the discussion (as well as based upon the supporting evidence that was provided). The project also followed the lead of other projects who had already gone through the same debate (particularly Drupal).

As for failing WCAG compliance, I have to disagree. Here are the tests from G130, the sufficient technique for headings with regards to the success criterion 2.4.6 (https://www.w3.org/TR/2015/NOTE-WCAG20-TECHS-20150226/G130):

1. Determine if the Web page contains headings.
2. Check that each heading identifies its section of the content.

Even if you are not in favour of "You are here", it still passes the tests as described in the sufficient technique for success criterion 2.4.6 as the heading does identify the section of content since the purpose of a breadcrumb is to identify where the user is located within the site hierarchy. The section of content is also further identified with the use of the <code>nav</code> element.

As for success criterion 1.3.1, here are the tests from H42, the sufficient technique for <code>h1</code> to <code>h6</code> headings for success criterion 1.3.1 (https://www.w3.org/TR/2015/NOTE-WCAG20-TECHS-20150226/G130):

1. Check that heading markup is used when content is a heading.
2. Check that heading markup is not used when content is not a heading.

The h2 is being used for "You are here" which is the heading for a section of content (which the section is further identified with the use of the <code>nav</code> element). You may disagree with the wording of the heading, but it still meets success criterion 1.3.1 because heading markup was used to programmatically identify a heading (which is associated to a section of content using the <code>nav</code> element). 


As for WebAim using "You Are Here", here is the relevant breadcrumb markup from WebAim (http://webaim.org/services/training/): 
```html
<p id="breadcrumbs"><span class="hidden">You are here: </span><a href="/">Home</a> &gt; <a href="/services/">Services</a> &gt; Accessibility Training</p>
````

Here is the breadcrumb markup from Drupal.org (https://www.drupal.org/drupalorg/blog/whats-new-on-drupalorg-february-2016):
```html
 <h2 class="element-invisible">You are here</h2><div class="breadcrumb"><a href="/">Home</a> » <a href="/drupalorg">Drupal.org</a> » <a href="/drupalorg/blog">Blog</a></div>
```

Here is the breadcrumb markup from the HTML5 spec:
````html
   <nav>
   <h2>You are here:</h2>
   <ul id="navlist">
   <li><a href="/">Main</a> →</li> 
   <li><a href="/products/">Products</a> →</li> 
   <li><a href="/products/dishwashers/">Dishwashers</a> →</li> 
   <li><a>Second hand</a></li> 
   </ul>
   </nav>
```

**Comment by shawnthompson** [comment]
@pjackson28, those examples are great for using the **You are here** text but what are your thoughts about putting the page title in that list?

The [Drupal](https://www.drupal.org/drupalorg/blog/whats-new-on-drupalorg-february-2016) example doesn't, which uses a heading for the **You are here**

![image](https://cloud.githubusercontent.com/assets/2575266/13641228/abe05552-e5e5-11e5-86ad-8784bd2966a1.png)

The [WebAim](http://webaim.org/services/training/) one does but isn't a heading for the **You are here**

![image](https://cloud.githubusercontent.com/assets/2575266/13641241/b9ad2598-e5e5-11e5-8360-c4208a32d3cf.png)

And neither of them are list items.

**Comment by dorina1** [comment]
@wet-boew/access-working-group I will have to agree with @Robineast on this: wording of the heading 2 "You are here" has absolutely no meaning to a screen reader user. Here where? Is "you are here" clear and descriptive so when we listen to it we will easily understand the content easily? 
Even if the <nav> tag is used before the invisible H2 "You are here", that does not give us any indication that in that navigation region is a breadcrumb trail. Also, when we navigate by headings, how is "you are here" offering any meaningful information regarding the content that is available under that heading?

**Comment by LaurentGoderre** [comment]
There is an inherent problem with choosing metaphors that will speak to everyone. Both "You are here" and "Breadcrumb Trail" are metaphors and whether it is clear or not is completely arbitrary.

This is not a failure of 2.4.6 Headings and Labels. However, there might be way to make the text more universally understood but that won't be easy.

**Comment by davidpepin** [comment]
I totally agree with @shawnthompson comment (and it seemed to go unnoticed): sinceTitle of the page you are on was removed from the breadcrumb list, the statement has become false. 

For example, on https://www.canada.ca/en/services/policing/police/aboriginal-policing.html, the trail says: 
You are here:
1.	Home
2.	Policing, justice and emergencies
3.	Policing

But that’s not true, you are not on Policing; you are one level below that. 

Same applied to all pages. On https://www.canada.ca/en/services/policing.html, the trail says: 
You are here:
1.	Home

But you’re not on the home page; you’re one level deeper.

**Comment by pjackson28** [comment]
The challenge we have here is we are now past what is required to meet accessibility and into what may be the most usable for a given user. The challenge is that this depends upon context, the user background, knowledge, etc. Even the accessibility community is divided upon what is the right approach as there is no clear right or wrong way. "You are here" works for some (including some screen readers) and not for others (including some screen readers). Same with "Breadcrumb trail", "Breadcrumb", and any other number of variations. The same was the case for whether to use a list or not, whether to include the page name in the breadcrumb, whether to use a heading and even whether to use a nav element. For me, Breadcrumb trail works because I understand the meaning but what about users who don't? What does breadcrumb trail mean to a user who doesn't know the various stories where it is used to find the way home?

That is why we went with the consensus decision at the time and to try to align with the accessibility work done by others (in particular the Drupal work which people involved with that actually triggered the debate in WET in the first place). The same arguments that have been raised in this thread (among many other arguments) were made for and against each of the approaches but in the end "You are here", in an h2, nested in a nav, with breadcrumb elements in a list without the page title in the breadcrumb was determined to be the better way forward from an accessibility and usability perspective by the majority of the group involved in the debate (which mostly consisted of people from the web accessibility community).  

@davidpepin @shawnthompson On most of the pages the h1 (the page title) is the next item after the breadcrumb. Not the same as including it in the breadcrumb trail but close. The reason it was taken out of the breadcrumb trail was not just because of duplication but because of the big issues on mobile where the breadcrumb in some cases took up over a third of the screen (exceeding 6 wrapped lines in some cases) making it a much less usable experience for mobile users. So that is where the consensus was that the impact of including the title in the breadcrumb trail was far more severe for mobile users than from a semantic standpoint with the page title removed.

**Comment by LaurentGoderre** [comment]
If it made more sense to have it there to understand the 'You are here', we could include it to the list but hide it so that only sccreen readers get that part.

**Comment by shawnthompson** [comment]
That's was my next suggestion.

**Comment by pjackson28** [comment]
@LaurentGoderre If I recall correctly, one of the reasons raised previously for having it as a heading was to provide a mechanism for the user to quickly get back to the breadcrumb trail (whether through screen reader navigation or through another tool that enables navigation according to the HMTL5 outline). If you remove the heading and put it in the breadcrumb trail then you don't have a easy way to quickly navigate there anymore. Also if you use a nav element then there is supposed to be a nested heading element so that the nav doesn't become an unnamed nav section in the HTML5 outline (which the nested heading element is also read off by certain screen readers, NVDA in particular, not sure if JAWS fixed that bug yet).

**Comment by LaurentGoderre** [comment]
@pjackson28I didn't mean change the heading, I meaning adding back the title of the page to the list of breadcrumb but hide it

**Comment by shawnthompson** [comment]
``` html
<nav role="navigation" id="wb-bc" property="breadcrumb">
<h2 class="wb-inv">You are here:</h2>
<div class="container">
<div class="row">
<ol class="breadcrumb">
	<li><a href='/en.html'>Home</a></li>
	<li><a href='/en/services/policing.html'>Policing, justice and emergencies</a></li>
	<li><a href='/en/services/policing/police.html'>Policing</a></li>
	<li class="wb-inv">Aboriginal policing</li>
</ol>
</div>
</div>
</nav>

...
 <h1 property="name" id="wb-cont">Aboriginal policing</h1> 

```

**Comment by pjackson28** [comment]
@LaurentGoderre Oh, in that case I don't think it would be an issue other than being a markup change. The only risk I see is the possibility that the h1 and the title in the breadcrumb could get out of sync because the title in the breadcrumb wouldn't be visible to authors. Not an issue if it is auto-generated but could potentially be an issue for manually maintained pages.

**Comment by LaurentGoderre** [comment]
Yeah, I thought about that too.... I guess those are the reason why for quality web content, a generator (like Assemble or Jekyll) or a CMS is a bug plus

**Comment by dorina1** [comment]
@wet-boew/access-working-group @pjackson28 You mentioned the followings: 
" Here are the tests from G130, the sufficient technique for headings with regards to the success criterion 2.4.6 (https://www.w3.org/TR/2015/NOTE-WCAG20-TECHS-20150226/G130):
    Determine if the Web page contains headings.
    **Check that each heading identifies its section of the content.**"

Yes, the page contains headings, but I still don't see how You are here identifies its section of the content when we navigate by headings and we try to make sense of the page layout/content.
This is no different than those "click here" links. 

WebAim is using "You Are Here" but in a <p> tag which makes a big difference for a screen user experience. We appreciate having an invisible heading and being able to jump to that section, but we need a proper heading to identify the breadcrumb navigation section.

**Comment by LaurentGoderre** [comment]
But the heading does identify it's section of content. The fact that there is disagreement on the best metaphor to use does not constitute a failure of that criterion.

**Comment by dfarough** [comment]
I agree that "You are here" is not the best text to be used for this heading.  Perhaps we should consult with Web usability experts to find a better term which adequately describes the purpose of the content. 

So far the only thing I can come up with is "Current Site Location"

I agree with @LaurentGoderre that this should not constitute a failure, however, I think we do need to make an effort to provide something that is more helpful to our users.
Maybe @laurawesley or @thomasgohard can help.

**Comment by LaurentGoderre** [comment]
100% agree with @dfarough ! Thanks for weighing in!

**Comment by hsrudnicki** [comment]
Please update this issue's status, whether this can be closed or otherwise? Thx
[ as well into consideration, maybe this shall be reflected/mentioned in #5574 ]

**Comment by shawnthompson** [comment]
This was brought up today in an Accessibility class and I was asked to revisit this issue. I am willing to make the change if we can agree on what to change it to in both official languages?

**Comment by rlambert27** [comment]
Yes the markup does match the G65 technique which is correct. However the text being used "You are here" is what is being disputed, not the markup. To someone who is visually impaired what does "You are here" mean, that's kind of like says "Look at this".

The text could be improved to something that will give a more meaningful context to the content in this section such as "Current site location".

I've been trying to find other GOV websites using a heading but so far no go. Most GOV websites who have breadcrumbs don't have heading for it. 

Research goes on....

**Comment by SBleeks** [comment]
The term most often used to describe this section is “Breadcrumb Navigation”. I think this would work well as a heading. I’m not sure what the French translation would be.

**Comment by rlambert27** [comment]
I think the official "Breadcrumb" translation in french is "Fil d'ariane". So it could be "Navigation du fil d'ariane". But I don't really like that term, some people may not know what "Fil d'ariane" is in french and "Breadcrumb" also sounds.......generic.

Maybe something like:

Site navigation location (or insert synonym for location) 
- area
- locale
- region
- section (kinda like this one)

In french:
Section de la navigation du site
Emplacement de la navigation du site

Just some thoughts :-)

**Comment by RobJohnston** [comment]
Indeed, calling it [Ariadne's thread](https://en.wikipedia.org/wiki/Ariadne%27s_thread_%28logic%29) in English would be a head-scratcher, but that's what it is and according to the [Translation Bureau](http://www.btb.termiumplus.gc.ca/tpv2alpha/alpha-eng.html?lang=eng&i=1&srchtxt=BREADCRUMB+TRAIL&index=alt&codom2nd_wet=1#resultrecs), these are also acceptable translations of "Breadcrumb trail":

- piste de navigation 
- chemin de navigation 
- chemin de fer

**Comment by dorina1** [comment]
@wet-boew/access-working-group  
AWG discussed and agreed to have the 'You are here" English heading and "Vous êtes ici :" French heading replaced with the following:
English - Breadcrumb Navigation 
French - Piste de navigation

**Comment by EricDunsworth** [comment]
@nschonni @LaurentGoderre @lucas-hay What are your thoughts with regards to the i18n versions of that heading?

Should the updated heading's content be translated into all of those languages before any attempt is made to update its French/English entries in [src/i18n/i18n.csv](https://github.com/wet-boew/wet-boew/blob/master/src/i18n/i18n.csv)?

**Comment by duboisp** [comment]
@lucas-hay let me know if you want me to update the i18n.

**Comment by shawnthompson** [comment]
LOL this was assigned to me. I just shared it in a thread. Maybe I should do this change.

Are you all still ok with this? @duboisp, @dorina1, @rlambert27?

(hashtag)better late than never

**Comment by dfarough** [comment]
Sounds fine to me.
would be nice to decide something

**Comment by shawnthompson** [comment]
English - Breadcrumb Navigation
French - Piste de navigation

**Comment by shawnthompson** [comment]
Ok after looking at this further I realized that this text has been translated into over 30 languages. 

Now I need to make a decision:

1. do I just change the French and English text and leave all the others?
2. do I use something like Google to translate all the other languages?
3. or something else I didn't think of?

Not a decision I want to make on my own so please weigh in folks.

**Comment by thekodester** [comment]
We really shouldn’t use google translate as it not that reliable still.

Sucks to say it but English and French are considered the primary language argument could be made to just remove that string from all others until a translate can be made if ever.

###### I am just a volunteer on this GitHub project.

**Comment by RobJohnston** [comment]
Deepl is better than Google, but I use Wikipedia pages to get good context translations like this.

**Comment by Robineast** [comment]
Categorization: Unclassified
Shawn, we should stick to what we know. In this case English and French unless Folks around the table know other languages. Do not try to translate other languages as it is not our role. Use our official languages only.

Note: ASL, QSL, ISL are primary sign languages in Canada. Officially adopted with the passing of the ACA on June 21, 2019. So stick to E and F or if you go with the other languages then the next three for translation are the three official Canadian Sign Languages before you get to the rest.

Robin and Lumos

**Comment by dfarough** [comment]
I agree with Robin, stick with the languages that we know. Anyone using wet using other languages would know the best translation to use for their language.

**Comment by thomasgohard** [comment]
I believe that, in the past, GAC, did some of the translations for languages other than English and French. Might be worth reaching out to them.

**Comment by shawnthompson** [comment]
What about crowd sourcing the other translations. I'm pretty sure originally we paid for the translations back when WET had money.

**Comment by dorina1** [comment]
> English - Breadcrumb Navigation
> French - Piste de navigation

Sounds good to me :) 
@shawnthompson Thanks for taking care of this.

**Comment by RobJohnston** [comment]
Another source is Microsoft's Language Portal.  For example,
https://www.microsoft.com/en-us/language/Search?&searchTerm=Breadcrumb%20Navigation&langID=736&Source=true&productid=undefined

**Comment by shawnthompson** [comment]
Please keep in mind this change is for the Web Experience Project, which is not just for the Government of Canada. It's an open source project used by many different sites in many different languages. 

This added a forth option, do we make this change only in GCWeb and keep it like this in WET?

**Comment by EricDunsworth** [comment]
> This added a forth option, do we make this change only in GCWeb and keep it like this in WET?

@shawnthompson Given how generic something like the breadcrumb trail's heading is, IMO it wouldn't be appropriate to only contribute it to GCWeb. Having user-friendly heading content shouldn't be specific to Canada.ca. Especially since GCWeb would normally inherit that content from WET.

As far as the i18n debacle goes... IIRC I was intending to send in a PR when I posted https://github.com/wet-boew/wet-boew/issues/7432#issuecomment-339459939, but no one followed-up at the time about how best to tackle the i18n strings. Plus it probably fell off my radar :P.

I don't like the idea of keeping the i18n strings as-is, but it's probably the lesser of two evils to have outdated translations as opposed to reverting them to fall-back on the new English string. Borrowing Microsoft's translations (if licensing terms allow it) might not hurt though...

WET's pre-existing English breadcrumb string is "You are here:" and the proposed string is "Breadcrumb navigation". Since Microsoft's strings use "Breadcrumb navigation" as a starting point, their translations would likely be a better fit for the direction you're going in. Even if the GC's own translators disagreed with some of Microsoft's translations, Microsoft's ones would probably still be truer the direction you're going in compared to what we have right now.

IMO your best course of action would be to send over a PR that updates the English/French strings at a minimum. That'll show that you're serious about getting the change implemented (especially since you're still this issue's assignee lol).

Good luck :)!
